test/powershell/Host/ConsoleHost.Tests.ps1

```diff
@@ -599,7 +599,7 @@ foo
                 $out = pwsh -workingdirectory ~ -c '(Get-Location).Path'
                 $out | Should -HaveCount 2
                 $out[0] | Should -BeExactly (Get-Item ~).FullName
-                $out[1] | Should -BeExactly "$testdrive"
+                $out[1] | Should -BeExactly [System.IO.FileInfo]::new($testdrive).FullName
```

Comment:
This should be put into a test helper module function, with a name that makes it clear it is converting to Windows long file name as needed.

Reply:
Need to think through what is the right thing to do here.  I don't like the idea of have this sprinkled throughout our test code.  Another option is to change how we get environment variables and always return the long path version since the 8.3 syntax is only relevant on downlevel Windows

Reply:
Might be better to have `Join-Path` always return a long path which would also address the test issue (or most of them).  Still trying to understand what changed in .Net Core 3.0 that's causing this

Reply:
I've decided that by default, the FileSystemProvider will return long paths